src/driver.ml

```diff
@@ -670,23 +618,13 @@ let with_preprocessed_file fn ~f =
     protectx (Caml.Filename.temp_file "ocamlpp" "")
       ~finally:remove_no_error
       ~f:(fun tmpfile ->
-        let comm =
-          Printf.sprintf "%s %s > %s"
-            pp (if String.equal fn "-" then "" else Caml.Filename.quote fn)
-            (Caml.Filename.quote tmpfile)
-        in
-        if Caml.Sys.command comm <> 0 then
-          raise (Pp_error { filename = fn
-                          ; command_line = comm
-                          });
-        f tmpfile)
-
-let with_preprocessed_input fn ~f =
-  with_preprocessed_file fn ~f:(fun fn ->
-    if String.equal fn "-" then
-      f stdin
-    else
-      In_channel.with_file fn ~f)
+        match System.run_preprocessor ~pp ~input:fn ~output:tmpfile with
+          | Ok () -> f tmpfile
+          | Error (failed_command, fall_back_version) ->
+            create_versioned_error_f fall_back_version fn ~f:(fun create_f ->
+            create_f "Error while running external preprocessor\n \
```

Comment:
```suggestion
            create_f "Error while running external preprocessor\n\
```

(I'm guessing this space wasn't intentional)

Reply:
Good catch